tools/preconf-rpc/sender/sender.go

```diff
@@ -528,7 +521,13 @@ func (t *TxSender) sendBid(
 		}
 	}
 
-	cost, blockNo, err := t.calculatePriceForNextBlock(txn, prices)
+	start := time.Now()
+	optedInSlot := timeToOptIn <= blockTime-(prices.MsSinceLastBlock/1000)
```

Comment:
Can't we directly check for the slot number? This seems a bit fragile with preicse ms timing etc.